src/components/ContactCard.vue

```diff
@@ -0,0 +1,239 @@
+<template>
+  <form class="contact-card__container px-4 needs-validation" novalidate>
+    <p class="contact-card__title iransans mb-4">
+      ما برای همکاری با شما آماده ایم؛ <br />
+      با ما در تماس باشید!
+    </p>
+    <input
+      v-model="name"
+      id="name"
+      type="text"
+      class="contact-card__input iransans form-control"
+      :class="(validationFeedbacks.name) ? 'not-valid' : ''"
+      placeholder="نام شما"
+      required
+      minlength="5"
+      @input="validateName"
+    />
+    <p class="contact-card__validation-feedback iransans">{{ validationFeedbacks.name }}</p>
+    <input
+      v-model="phoneNumber"
+      id="phoneNumber"
+      type="text"
+      class="contact-card__input mt-4 iransans form-control"
+      :class="(validationFeedbacks.phoneNumber) ? 'not-valid' : ''"
+      placeholder="شماره تماس"
+      @input="validatePhoneNumber"
+    />
+    <p class="contact-card__validation-feedback iransans">{{ validationFeedbacks.phoneNumber }}</p>
+    <textarea
+      v-model="message"
+      id="message"
+      class="contact-card__textarea form-control mt-4 iransans"
+      :class="(validationFeedbacks.message) ? 'not-valid' : ''"
+      placeholder="پیام شما"
+      rows="6"
+      @input="validateMessage"
+    ></textarea>
+    <p class="contact-card__validation-feedback iransans">{{ validationFeedbacks.message }}</p>
+    <button
+      @click="submit($event)"
+      type="submit"
+      class="
+        contact-card__submit
+        py-2
+        mt-4
+        d-flex
+        flex-row
+        justify-content-center
+      "
+    >
+      <span v-if="loading" class="spinner-border text-secondary" role="status" aria-hidden="true"></span>
+      <p v-if="!loading" class="iransans mx-1 my-auto">ارسال پیام</p>
+      <span v-if="!loading" class="mdi mdi-arrow-left-bold" style="color: black"></span>
+    </button>
+  </form>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  props: {
+    loading: {
+      type: Boolean,
+      default: false,
+    }
+  },
+  emits: ['submit'],
+  data() {
+    return {
+      name: '',
+      phoneNumber: '',
+      message: '',
+      validationFeedbacks: {
+        name: '',
+        phoneNumber: '',
+        message: ''
+      },
+      isValid: false,
+    }
+  },
+  methods: {
+    submit(event: MouseEvent) {
+      event.preventDefault()
+      this.validateName()
+      this.validatePhoneNumber()
+      this.validateMessage()
+      if (this.isValid) {
+        this.$emit('submit', {
+          name: this.name,
+          phoneNumber: this.phoneNumber,
+          message: this.message
+        })
+        this.clearInputs()
```

Comment:
if something goes wrong (for example server is down) all inputted data by user will cleared.
it's a very poor UX and can easily make user angry.

Reply:
Fixed: clearing inputs only on success